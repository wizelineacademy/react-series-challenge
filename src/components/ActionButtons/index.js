import AddFavButton from './addFavButton';
import RemoveFavButton from './removeFavButton';

export {
    AddFavButton,
    RemoveFavButton
}
