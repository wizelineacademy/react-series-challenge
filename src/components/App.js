import React from "react";
import { Route, Switch, Redirect } from "react-router-dom";
import HomePage from "./pages/HomePage/index";
import FavsPage from "./pages/FavsPage/index";

const App = () => {
  return (
    <Switch>
      <Route exact path="/" render={() => <Redirect to="/home" />} />
      <Route path="/home" component={HomePage} />;
      <Route path="/favorites" component={FavsPage} />;
    </Switch>
  );
};

export default App;
