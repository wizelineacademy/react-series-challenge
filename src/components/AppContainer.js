import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Route, Switch } from 'react-router-dom';
import { bindActionCreators } from 'redux';

import gifActions from '../actions/gifActions';

import GifGrid from '../components/GifGrid';
import SearchBar from '../components/SearchBar';

import { ContainerWrapper } from '../styles/ContainerWrapper.style.js';


export class AppContainer extends Component {
  componentDidMount() {
    const { getTrendingGifs } = this.props;

    getTrendingGifs();
  }

  render() {
    const { gifs, favorites } = this.props;

    return (
      <ContainerWrapper className="mt">
        <SearchBar />

        <Switch>
          <Route
            exact
            path="/"
            render={({ match }) => (
              <GifGrid gifs={ gifs } />
            )}
          />
          <Route
            path="/favorites"
            render={({ match }) => (
              <GifGrid gifs={ favorites } />
            )}
          />
        </Switch>
      </ContainerWrapper>
    );
  }
}

const mapStateToProps = (state) => state;

const mapDispatchToProps = (dispatch) => {
  const { getTrendingGifs } = gifActions.creators;

  return bindActionCreators({ getTrendingGifs }, dispatch);
};

export default connect(mapStateToProps, mapDispatchToProps)(AppContainer);
