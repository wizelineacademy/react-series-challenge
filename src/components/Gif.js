import React from 'react';
import styled from 'styled-components';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { toggleFavorite } from '../actions/favorites';
import { isFavorited } from '../selectors/favorites';
import { selectedGifChange } from '../actions/selected';

const FAVORITE_ICON = '⭐';

export const StyledGif = styled.img`
  width: 100%;
  height: auto;
  display: block;
`;

export const StyledFavoritedIcon = styled.a`
  cursor: pointer;
  transition: filter .15s ease;
  content: '${FAVORITE_ICON}';
  position: absolute;
  bottom: 10px;
  right: 10px;
  ${({ active }) => active ? '' : `filter: grayscale(100%);`}

  &:hover: {
    text-shadow: -1px 0px 4px rgba(20, 20, 20, 1);
  }
`;

export const Gif = ({ big, gif, favorited, selectedGifChange, toggleFavorite }) => {
  const src = !big
    ? gif.images.fixed_width_downsampled.url
    : gif.images.original.url;
  return (
    <React.Fragment>
      <StyledFavoritedIcon onClick={() => toggleFavorite(gif)} active={favorited}>
        {FAVORITE_ICON}
      </StyledFavoritedIcon>
      <Link to={`?gifId=${gif.id}`}>
        <StyledGif src={src} onClick={() => selectedGifChange(gif)} />
      </Link>
    </React.Fragment>
  );
};

const mapStateToProps = (state, { gif }) => ({
  favorited: isFavorited(gif.id, state)
});

const mapDispatchToProps = {
  toggleFavorite,
  selectedGifChange
};

export default connect(mapStateToProps, mapDispatchToProps)(Gif);
