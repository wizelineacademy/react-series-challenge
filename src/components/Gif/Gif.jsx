import React from 'react';
import PropTypes from 'prop-types';

import {
  GifContainer,
  GifTitle,
  GifImage,
  GifLikeContainer,
} from './Gif.style';

import likeContainer from './favorite-heart-like-container.png';
import likeSelected from './favorite-heart-like-selected.png';

const Gif = ({ object, addFavorite, removeFavorite }) => (
  <GifContainer>
    <GifTitle>{object.title}</GifTitle>
    <GifImage src={object.embed_url} title={object.title} />
    {
      object.isFavorite ?
        (
          <GifLikeContainer
            src={likeSelected}
            alt="Remove from favorite"
            onClick={() => removeFavorite({ payload: { gif: object } })}
          />
        ) :
        (
          <GifLikeContainer
            src={likeContainer}
            alt="Add to favorite"
            onClick={() => addFavorite({ payload: { gif: object } })}
          />
        )
    }
  </GifContainer>
);

Gif.propTypes = {
  object: PropTypes.shape({
    embed_url: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    id: PropTypes.string.isRequired,
    isFavorite: PropTypes.bool.isRequired,
  }).isRequired,
  addFavorite: PropTypes.func.isRequired,
  removeFavorite: PropTypes.func.isRequired,
};

export default Gif;

