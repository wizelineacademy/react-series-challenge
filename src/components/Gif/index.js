import { bindActionCreators } from 'redux';
import { connect } from 'react-redux'
import gif from '../../actions/gif'
import React from 'react'

export const Gif = ({ src, alt, onClick, className, gif, favorite }) => {
    return <img alt={alt} src={src} onClick={onClick.bind(null, gif)} className={ favorite ? `${className} favorite` : className } />
}


const mapDispatchToProps = dispatch => {
    /**
     *
     */
    const { creators: { clickGif: onClick } } = gif
    /**
     *
     */
    return bindActionCreators({ onClick }, dispatch)
}

export default connect(null, mapDispatchToProps)(Gif)