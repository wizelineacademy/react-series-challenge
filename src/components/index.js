import Card from './Card'
import Detail from './Detail'
import GifItem from './GifItem'
import Grid from './Grid'
import Header from './Header'
import Home from './Home'
import SearchBar from './SearchBar'

export {
  Card,
  Detail,
  GifItem,
  Grid,
  Header,
  Home,
  SearchBar
}
