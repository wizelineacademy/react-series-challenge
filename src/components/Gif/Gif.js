import React from 'react';
import {MasonryItem, Overlay, ImageWrapper, LinkStyled, GifImg, GifTitle} from "./Gif.styled";

const getPreviewGifUrl = (gif) => {
  if (gif.images && gif.images.original) {
    return gif.images.original.url;
  } else {
    return '';
  }
};

const handleFavButton = (gif, addGif, removeGif, isFavorited) => {
  if (!isFavorited) {
    addGif(gif);
  } else {
    removeGif(gif.id);
  }
};

export const Gif = ({ gif, fullWidth, addGif, removeGif, isFavorited, isInDetail }) => (
  <MasonryItem fullWidth={fullWidth}>
      <ImageWrapper>
        <GifImg isFavorited={isFavorited} src={getPreviewGifUrl(gif)} alt={gif.title} />
        <Overlay>
          <GifTitle>Title: {gif.title}</GifTitle>
          <button id="fav-btn" onClick={() => handleFavButton(gif, addGif, removeGif, isFavorited)}>
            {!isFavorited ? 'Add to Favorites': 'Remove from favorites'}
          </button>
          { !isInDetail ? (
            <LinkStyled to={{pathname: `/detail/${gif.id}`}}>
              View detail
            </LinkStyled>) : ''
          }
        </Overlay>
      </ImageWrapper>
  </MasonryItem>
)