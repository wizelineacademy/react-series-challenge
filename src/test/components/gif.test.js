import { Gif } from '../../components/Gif'
import { shallow, simulate } from 'enzyme'
import React from 'react';

describe('Gif', () => {
    let testValue = false;
    const gif = {
        src: 'https://media3.giphy.com/media/UdtJqVG2gCchy/giphy-downsized.gif',
        alt: 'BMO',
        onClick: () => { testValue = true },
        favorite: false
    }


    test('Should match with a snapshot', () => {
        const component = shallow(<Gif {...gif} />)
        expect(component).toMatchSnapshot();
    })


    test('Should not have a favorite class', () => {
        const component = shallow(<Gif {...gif} />)
        expect(component).toMatchSnapshot();
        expect(component.hasClass('favorite')).toBeFalsy()
    })


    test('Should match with a snapshot', () => {
        const liked = Object.assign({}, gif, { favorite: true })
        const component = shallow(<Gif {...liked}  />)
        expect(component).toMatchSnapshot();
    })

    test('Should have a favorite class', () => {
        const liked = Object.assign({}, gif, { favorite: true })
        const component = shallow(<Gif {...liked} />)
        expect(component.hasClass('favorite')).toBeTruthy()
    })

    test('Should execute the click', () => {
        const liked = Object.assign({}, gif, { favorite: true })
        const component = shallow(<Gif {...liked} />)
        component.simulate('click')
        expect(testValue).toBeTruthy()
        testValue = false;
    })
    
});