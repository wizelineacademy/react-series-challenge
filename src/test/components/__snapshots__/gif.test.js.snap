// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gif Should match with a snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    alt="BMO"
    favorite={false}
    onClick={[Function]}
    src="https://media3.giphy.com/media/UdtJqVG2gCchy/giphy-downsized.gif"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "alt": "BMO",
      "className": undefined,
      "onClick": [Function],
      "src": "https://media3.giphy.com/media/UdtJqVG2gCchy/giphy-downsized.gif",
    },
    "ref": null,
    "rendered": null,
    "type": "img",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "alt": "BMO",
        "className": undefined,
        "onClick": [Function],
        "src": "https://media3.giphy.com/media/UdtJqVG2gCchy/giphy-downsized.gif",
      },
      "ref": null,
      "rendered": null,
      "type": "img",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Gif Should match with a snapshot 2`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    alt="BMO"
    favorite={true}
    onClick={[Function]}
    src="https://media3.giphy.com/media/UdtJqVG2gCchy/giphy-downsized.gif"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "alt": "BMO",
      "className": "undefined favorite",
      "onClick": [Function],
      "src": "https://media3.giphy.com/media/UdtJqVG2gCchy/giphy-downsized.gif",
    },
    "ref": null,
    "rendered": null,
    "type": "img",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "alt": "BMO",
        "className": "undefined favorite",
        "onClick": [Function],
        "src": "https://media3.giphy.com/media/UdtJqVG2gCchy/giphy-downsized.gif",
      },
      "ref": null,
      "rendered": null,
      "type": "img",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Gif Should not have a favorite class 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    alt="BMO"
    favorite={false}
    onClick={[Function]}
    src="https://media3.giphy.com/media/UdtJqVG2gCchy/giphy-downsized.gif"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "alt": "BMO",
      "className": undefined,
      "onClick": [Function],
      "src": "https://media3.giphy.com/media/UdtJqVG2gCchy/giphy-downsized.gif",
    },
    "ref": null,
    "rendered": null,
    "type": "img",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "alt": "BMO",
        "className": undefined,
        "onClick": [Function],
        "src": "https://media3.giphy.com/media/UdtJqVG2gCchy/giphy-downsized.gif",
      },
      "ref": null,
      "rendered": null,
      "type": "img",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
